### PR TITLE
Update to 2.1.0

### PR DESCRIPTION
**Pull request by cmlsharp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,15 +24,16 @@ optional arguments:
 
 `style50` takes zero or more files/directories to style check. If no arguments are given, `style50` will recursively check the current directory. 
 
-`MODE` can be one of `side_by_side` (default), `unified`, `raw`, and `json`. `side_by_side` and `unified` output side-by-side and unified diffs between the inputted file and the correctly styled version respectively. `raw` outputs the raw percentage of correct (unchanged) lines, while `json` outputs a json object containing information pertinent to the CS50 IDE plugin (coming soon).
+`MODE` can be one of `character` (default), `side_by_side`, `unified`, `raw`, and `json`. `character`, `side_by_side`, and `unified` output character-based, side-by-side, and unified diffs between the inputted file and the correctly styled version respectively. `raw` outputs the raw percentage of correct (unchanged) lines, while `json` outputs a json object containing information pertinent to the CS50 IDE plugin (coming soon).
```
Comment (dmalan) [inline README.md]: Is `raw` better described as `score`? Is there something better than `side_by_side`? That'd be annoying to type.

File: README.md [hunk 2]
```diff
@@ -24,15 +24,16 @@ optional arguments:
...
-    * C++
-    * Python
-    * Javascript
-    * Java
+
+* C
+* C++
+* Python
+* Javascript
+* Java
```
Comment (dmalan) [inline README.md]: Stylistically, let's start using hyphens for our Markdown lists, for consistency.

File: style50/checks.py [hunk 3]
```diff
@@ -51,6 +51,11 @@ def count_comments(self, code):
             prev_type = t_type
         return comments
 
+    def count_lines(self, code):
+        # count_lines removes blank lines by default,
```
Comment (dmalan) [inline style50/checks.py]: Should be docstring-style comment, no?

File: style50/checks.py [hunk 4]
```diff
@@ -51,6 +51,11 @@ def count_comments(self, code):
...
+        # but blank lines are relavent to style per pep8
+        return len(code.splitlines())
+
     # TODO: Determine which options (if any) should be passed to autopep8
     def style(self, code):
         return autopep8.fix_code(code, options={"max_line_length": 100})
```
Comment (dmalan) [inline style50/checks.py:63]: Interesting that they use a `dict` instead of named args.

File: style50/core.py [hunk 5]
```diff
@@ -54,7 +54,7 @@ class Style50(object):
     """
```
Comment (dmalan) [inline style50/core.py:54]: s/which/that/

File: style50/core.py [hunk 6]
```diff
@@ -54,7 +54,7 @@ class Style50(object):
...
     extension_map = {}
```
Comment (dmalan) [inline style50/core.py]: Document what this is.

File: style50/core.py [hunk 7]
```diff
@@ -64,35 +64,47 @@ def __init__(self, paths, output="side-by-side"):
...
+                print(footer, end="")
 
+            print(header.format(file), end="")
             try:
                 results = self._check(file)
             except Error as e:
                 termcolor.cprint(e.msg, "yellow", file=sys.stderr)
                 continue
 
             if results.diffs:
```
Comment (dmalan) [inline style50/core.py]: Do comment all sizable blocks of code.

File: style50/core.py [hunk 8]
```diff
@@ -167,26 +179,44 @@ def side_by_side(old, new):
...
             if diff[0] == " ":
                 yield diff
             elif diff[0] == "?":
                 continue
             else:
                 yield termcolor.colored(diff, "red" if diff[0] == "-" else "green", attrs=["bold"])
 
+    def html_diff(self, old, new):
+        """
+        Return HTML formatted character-based diff between old and new (used for IDE).
```
Comment (dmalan) [inline style50/core.py]: CS50 IDE

File: style50/core.py [hunk 9]
```diff
@@ -197,20 +227,17 @@ def _html_diff(old, new):
...
+                yield fmt("".join(buffer), dtype)
                 dtype = d[0]
-                buffer.clear()
+                buffer = []
 
             if dtype is None:
                 break
-            buffer.append(d[2:])
+
+            # Show insertions/deletions of whitespace clearly
```
Comment (dmalan) [inline style50/core.py:242]: Good comment.

**Review by dmalan — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (cmlsharp) [reply README.md]: I like `score`. `split` instead of `side_by_side`?

**Comment by dmalan** [comment]
+1

File: style50/core.py (hunk 5 above)
Comment (cmlsharp) [reply style50/core.py:54]: Fixed

File: style50/checks.py (hunk 4 above)
Comment (cmlsharp) [reply style50/checks.py:63]: The main function likely just hands it the object outputted by `optparse` or similar?

**Review by dmalan — APPROVED** [review]

**Pull request merged**